### DISCUSSION TITLE
How to only build graph without starting the service

### DISCUSSION THREAD

**Discussion by mabilinab** [gis-ops/docker-valhalla#156] [Q&A] [answered]
Hi everybody,
How can I run the docker image in a way that it only builds the graph without starting the valhalla service?
For example, I want it to stop at this step
```
"INFO: Successfully built files: /custom_files/corse-latest.osm.pbf
INFO: Hashing files: /custom_files/corse-latest.osm.pbf
INFO: Hashing file: /custom_files/corse-latest.osm.pbf
e30d46d20cfb405d63a53939cd47dd6b2c96dfce1540ae41ef21527e36fb5fb5
2024-05-18 10:05:46,465  INFO: Finished tarring 137 tiles to /custom_files/valhalla_tiles.tar
WARNING: User valhalla is running with sudo privileges. Try building the image with a host user's UID & GID."
```
And not go into:
`"INFO: Found config file. Starting valhalla service!"`

I'm asking this because I am using a kubernetes job to solely build the graph and then exiting – then I'll restart the other k8 worker pods to use the newly built graph from the k8 job. 
The issue is that by default, the docker container builds the graph then starts the valhalla server and keeps listening and therefore the k8 job never stops.
Thank you!

**Comment by nilsnolde** [comment, marked as answer]
https://github.com/gis-ops/docker-valhalla?tab=readme-ov-file#environment-variables

**Reply by mabilinab** [reply]
Oh shoot I missed the "serve_tiles".
Thank you Nils!
